src/bridge/AdminRoomHandler.ts

```diff
@@ -419,8 +419,10 @@ export class AdminRoomHandler {
                 notice = new MatrixAction(
                     "notice", `Successfully stored password for ${domain}. You will now be reconnected to IRC.`
                 );
-                if (client) {
-                    await client.disconnect("iwantoreconnect", "authenticating", false);
+                for (let i = 0; i < clientList.length; i++) {
```

Comment:
Please use `for (const client of clientList)`. You can then use `client.disconnect(` below.